packages/react/src/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,211 @@
+}
+
+// map tooltip direction to anchoredPosition props
+const directionToPosition: Record<TooltipDirection, {side: AnchorSide; align: AnchorAlignment}> = {
+  n: {side: 'outside-top', align: 'center'},
+  e: {side: 'outside-right', align: 'center'},
+  s: {side: 'outside-bottom', align: 'center'},
+  w: {side: 'outside-left', align: 'center'},
+}
+
+// map anchoredPosition props to tooltip direction
+const positionToDirection: Record<string, TooltipDirection> = {
+  'outside-top-center': 'n',
+  'outside-right-center': 'e',
+  'outside-bottom-center': 's',
+  'outside-left-center': 'w',
+}
+
+// The list is from GitHub's custom-axe-rules https://github.com/github/github/blob/master/app/assets/modules/github/axe-custom-rules.ts#L3
+const interactiveElements = [
+  'a[href]',
+  'button:not(:disabled)',
+  'summary',
+  'select',
+  'input:not([type=hidden])',
+  'textarea',
+]
+
+const isInteractive = (element: HTMLElement) => {
+  return (
+    interactiveElements.some(selector => element.matches(selector)) ||
+    (element.hasAttribute('role') && element.getAttribute('role') === 'button')
+  )
+}
+export const TooltipContext = React.createContext<{tooltipId?: string}>({})
+
+export const Tooltip = React.forwardRef(
+  ({direction = 's', text, type = 'description', children, id, className, ...rest}: TooltipProps, forwardedRef) => {
+    const tooltipId = useId(id)
+    const child = Children.only(children)
+    const triggerRef = useProvidedRefOrCreate(forwardedRef as React.RefObject<HTMLElement>)
+    const tooltipElRef = useRef<HTMLDivElement>(null)
+
+    const [calculatedDirection, setCalculatedDirection] = useState<TooltipDirection>(direction)
+
+    const openTooltip = () => {
+      if (tooltipElRef.current && triggerRef.current && !tooltipElRef.current.matches(':popover-open')) {
+        tooltipElRef.current.showPopover()
+      }
+    }
+    const closeTooltip = () => {
+      if (tooltipElRef.current && triggerRef.current && tooltipElRef.current.matches(':popover-open')) {
+        tooltipElRef.current.hidePopover()
+      }
+    }
+
+    useEffect(() => {
+      if (!tooltipElRef.current || !triggerRef.current) return
+      /*
+       * ACCESSIBILITY CHECKS
+       */
+      // Has trigger element or any of its children interactive elements?
+      const isTriggerInteractive = isInteractive(triggerRef.current)
+      const triggerChildren = triggerRef.current.childNodes
+      const hasInteractiveChild = Array.from(triggerChildren).some(triggerChild => {
+        return triggerChild instanceof HTMLElement && isInteractive(triggerChild)
+      })
+
+      if (
+        !(isTriggerInteractive || hasInteractiveChild) &&
+        (process.env.NODE_ENV === 'development' || process.env.NODE_ENV === 'test')
+      ) {
+        throw new Error(
+          'The `Tooltip` component expects a single React element that contains interactive content. Consider using a `<button>` or equivalent interactive element instead.',
```

Comment:
This feels more like a warning than an error. Curious if we want to use a console.warn here, wrapped in a guard to only appear in non-prod environments.